### PR TITLE
Updates app configs to use Figaro

### PR DESCRIPTION
**Pull request by amoose**
Continuing @jessieay's work to update the app configs...

### Why
The Dashboard configuration was spread across secrets.yml, saml.yml and application.yml

### How
Simplifies configurations to application.yml (Figaro) like other applications
Prepends all SAML configs with `saml_`
Coincides with [cookbook update PR](https://github.com/18F/identity-devops/pull/258)

Tested in `PT` 🏁 https://dashboard.pt.login.gov/

### CODE REVIEW

File: config/initializers/figaro.rb [hunk 1]
```diff
@@ -1,14 +1,9 @@
...
-  'IDP_SSO_URL',
-  'LOGIN_DASHBOARD_TOKEN',
-  'NEW_RELIC_LICENSE_KEY',
-  'SMTP_ADDRESS',
-  'SMTP_DOMAIN',
-  'SMTP_PASSWORD',
-  'SMTP_USERNAME',
-  'SP_NAME',
-  'SP_PASS',
+  'saml_idp_fingerprint',
```
Comment (zachmargolis) [inline config/initializers/figaro.rb]: Did we already commit a version of these that's lowercase? We kept the casing upper to match existing env vars in https://github.com/18F/identity-dashboard/pull/70#issuecomment-283431625 on purpose

Comment (zachmargolis) [inline config/initializers/figaro.rb]: nevermind, I just saw the "Coincides with cookbook update PR" part. Hooray integration!

File: config/application.yml.example [hunk 2]
```diff
@@ -1,45 +1,117 @@
...
+  dashboard_api_token: 'sekret'
+  idp_sp_url: 'http://localhost:3000/api/service_provider'
+  mailer_domain: 'https://dashboard.login.gov'
+  newrelic_license_key: ''
+  saml_idp_fingerprint: '8B:D5:C2:E8:9A:2B:CE:B7:4B:95:50:BA:16:79:05:27:17:D1:D3:67'
+  saml_idp_sso_url: 'http://localhost:3000/api/saml/auth'
+  saml_idp_slo_url: 'http://localhost:3000/api/saml/logout'
+  saml_name_id_format: 'urn:oasis:names:tc:SAML:1.1:nameid-format:persistent'
+  saml_sp_issuer: 'https://dashboard.login.gov'
+  saml_sp_certificate: |
```
Comment (zachmargolis) [inline config/application.yml.example]: would it simplify things on the chef side if we made these paths to files on the filesystem instead?

File: lib/saml_config.rb [hunk 3]
```diff
@@ -2,20 +2,18 @@
...
-      config_file = "#{Rails.root}/config/saml.yml"
-      saml_env_config = Hashie::Mash.new(YAML.load_file(config_file).fetch(Rails.env, {}))
       Hashie::Mash.new(
-        issuer: Figaro.env.SP_ISSUER || saml_env_config[:sp_issuer],
-        idp_sso_target_url: Figaro.env.IDP_SSO_URL || saml_env_config[:sso_url],
-        idp_slo_target_url: Figaro.env.IDP_SLO_URL || saml_env_config[:slo_url],
-        idp_cert_fingerprint: Figaro.env.IDP_FINERPRINT || saml_env_config[:idp_fingerprint],
-        name_identifier_format: Figaro.env.IDP_NAME_ID_FORMAT || saml_env_config[:name_id_format],
-        certificate: Figaro.env.SP_CERTIFICATE || saml_env_config[:sp_certificate],
+        issuer:                     Figaro.env.saml_sp_issuer,
```
Comment (jessieay) [inline lib/saml_config.rb]: we haven't been using this style of lining up vals, but no biggie, can clean up l8r

File: spec/mailers/user_mailer_spec.rb [hunk 4]
```diff
@@ -19,8 +19,8 @@
     let(:mail) { UserMailer.admin_new_service_provider(app) }
 
     it 'sends to ADMIN_EMAIL' do
-      ClimateControl.modify ADMIN_EMAIL: 'identity-admin@example.com' do
-        expect(mail.to).to eq ['identity-admin@example.com']
+      ClimateControl.modify ADMIN_EMAIL: 'partners@login.gov' do
```
Comment (jessieay) [inline spec/mailers/user_mailer_spec.rb]: do we have to change this to a potentially real email? I think we were trying to be compliant with https://tools.ietf.org/html/rfc2606 here

File: spec/requests/service_providers_spec.rb [hunk 5]
```diff
@@ -22,7 +22,7 @@
     end
 
     it 'allows admin to approve' do
-      ClimateControl.modify ADMIN_EMAIL: 'identity-admin@example.com' do
+      ClimateControl.modify ADMIN_EMAIL: 'partners@login.gov' do
```
Comment (jessieay) [inline spec/requests/service_providers_spec.rb]: same comment re: moving away from `example.com` https://tools.ietf.org/html/rfc2606

File: spec/mailers/user_mailer_spec.rb (hunk 4 above)
Comment (amoose) [inline spec/mailers/user_mailer_spec.rb]: right on, I'll update 👍

**Comment by amoose** [comment]
Updated per comments. Please review

**Comment by jessieay** [comment]
These changes look good, but when I do a search for `env` in the codebase I am still seeing results for `Figaro.env` with all caps in these files:

`config/environments/production.rb`
`config/environments/development.rb`
`config/database.yml`
`config/initializers/assets.rb`

should these be changed too?

**Comment by jessieay** [comment]
updated at  e75cf3ff141d9943bad5234675f81b6f92106f25 cc @zachmargolis

**Comment by jessieay** [comment]
Cool @amoose how should we coordinate merging this and the association cookbook PR so as to avoid prod issues?

**Comment by amoose** [comment]
Excellent! Merging! 

@jessieay Today, we will merge this and the [DevOps PR](https://github.com/18F/identity-devops/pull/258) at the same time. 

In the future, we will correlate application config updates with version bumps in the application which will require a corresponding change in the cookbook. see: https://github.com/18F/identity-private/issues/1709

**Pull request merged**